kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/util/IOUtil.java

```diff
@@ -8,114 +8,84 @@
  *
  * Contributors:
  *     Eurotech
+ *     Red Hat Inc
  *******************************************************************************/
 package org.eclipse.kura.core.util;
 
-import java.io.BufferedReader;
+import static java.lang.Thread.currentThread;
+
 import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.StringWriter;
 import java.net.URL;
 
+import org.apache.commons.io.IOUtils;
 import org.osgi.framework.Bundle;
 import org.osgi.framework.BundleContext;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-public class IOUtil {
+/**
+ * A few basic IO helper methods
+ */
+public final class IOUtil {
 
-    private static final Logger s_logger = LoggerFactory.getLogger(IOUtil.class);
+    private IOUtil() {
+    }
 
     /**
      * Reads a resource fully and returns it as a string.
      * 
-     * @param ctx
      * @param resourceName
-     * @return
+     *            the name of the resource
+     * @return the content as string, or {@code null if the resource coude not be found}
      * @throws IOException
+     *             in case there is a resource but it cannot be read
      */
-    public static String readResource(String resourceName) throws IOException {
-        URL resourceUrl = Thread.currentThread().getContextClassLoader().getResource(resourceName);
-        if (resourceUrl == null) {
-            return null;
-        }
-        return readResource(resourceUrl);
+    public static String readResource(final String resourceName) throws IOException {
+        return readResource(currentThread().getContextClassLoader().getResource(resourceName));
     }
 
     /**
      * Reads a resource fully and returns it as a string.
      * 
      * @param ctx
+     *            the bundle context to use for locating the resource
      * @param resourceName
-     * @return
+     *            the name of the resource
+     * @return the content as string, or {@code null if the resource coude not be found}
      * @throws IOException
+     *             in case there is a resource but it cannot be read
      */
     public static String readResource(BundleContext ctx, String resourceName) throws IOException {
-        URL resourceUrl = ctx.getBundle().getResource(resourceName);
-        if (resourceUrl == null) {
-            return null;
-        }
-
-        String resource = readResource(resourceUrl);
-        return resource;
+        return readResource(ctx.getBundle().getResource(resourceName));
     }
 
     /**
      * Reads a resource fully and returns it as a string.
      * 
-     * @param ctx
+     * @param bundle
+     *            the bundle to use for getting the bundle context
```

Comment:
In the implementation the `BundleContext` is never used. Broadly speaking, in the implementation, there is no such call (`bundle.getContext()`).